src/index.ts

```diff
@@ -4,16 +4,13 @@ export default function useLocalStorage(
   key: string,
   initialValue: string = ''
 ): [string, Dispatch<string>] {
-  const [item, setValue] = React.useState(() => {
-    const value = localStorage.getItem(key) || initialValue;
-    localStorage.setItem(key, value);
-    return value;
-  });
+  const [value, setValue] = React.useState(
+    () => window.localStorage.getItem(key) || initialValue
+  );
 
-  const setItem = (newValue: string) => {
-    setValue(newValue);
-    window.localStorage.setItem(key, newValue);
-  };
+  React.useEffect(() => {
+    window.localStorage.setItem(key, value);
+  }, [value]);
```

Comment:
Just curious, would including `key` in the dep make sense? 
As an array instance returned by `useLocalStorage` works per `key`, it might or might not be worth it.
Curious what you think.

Reply:
Yeap, it make sense. I'll include into deps.

I was also thinking if user changes the key, it should remove old storage key's value?

Reply:
Ah, I see what happened.

It was initially created so you'd need one `useLocalStorage` per key, thus `key` is not needed in the dep, as it'd never change.

It was to work similar to how `useState` works per state.

So never mind about adding the key to the dep list :)

Reply:
Okay, no problem.

Reply:
`react-hooks/exhaustive-deps` eslint says it otherwise... 
It looks like we'd need the `key` as a dep (or remove the deps array completey).

Sorry to trouble you again, as I was being ignorant not using the eslint check.
and would you add the `key` as one of the deps?

I will merge as soon as it's added to the deps list.

Thanks~

![image](https://user-images.githubusercontent.com/8465237/59235010-5b65f880-8bbd-11e9-83b8-bf75fb14e3f5.png)